libraries/chain/kv_context_rocksdb.cpp

```diff
@@ -0,0 +1,315 @@
+#include <eosio/chain/combined_database.hpp>
+#include <eosio/chain/controller.hpp>
+#include <eosio/chain/exceptions.hpp>
+#include <eosio/chain/kv_chainbase_objects.hpp>
+#include <eosio/chain/kv_context.hpp>
+
+namespace eosio { namespace chain {
+   struct kv_iterator_rocksdb : kv_iterator {
+      uint32_t&                num_iterators;
+      b1::chain_kv::view&          view;
```

Comment:
This seems like a bug waiting to happen.  Are there guarantees that the view will never go out of scope before the iterator?  Should we use some sort of a stronger reference such as weak_ptr/shared_ptr?  Or will that add too much overhead?